assets/js/hooks/session.js

```diff
@@ -979,6 +983,65 @@ const Session = {
     }
   },
 
+  toggleCollapseSection() {
+    if (this.focusedId) {
+      const sectionId = this.getSectionIdByFocusableId(this.focusedId);
+      const section = this.getSectionById(sectionId);
+
+      if (section) {
+        if (section.hasAttribute("data-js-collapsed")) {
+          section.removeAttribute("data-js-collapsed");
+        } else {
+          section.setAttribute("data-js-collapsed", "");
+          this.setFocusedEl(sectionId, { scroll: true });
+        }
+      }
+    }
+  },
+
+  toggleCollapseAllSections() {
+    const allCollapsed =
+      this.getSections().every(section => {
+        if (section.hasAttribute("data-js-collapsed")) {
+          return true;
+        } else {
+          return false;
+        }
+      });
+
+    if (allCollapsed) {
+      this.expandAllSections();
+    } else {
+      this.collapseAllSections();
+    }
+  },
+
+  expandAllSections() {
+    this.getSections().forEach(section => {
+      section.removeAttribute("data-js-collapsed");
+    });
+    if (this.focusedId) {
+      const focusedSectionId = this.getSectionIdByFocusableId(this.focusedId);
+
+      if (focusedSectionId) {
+        this.setFocusedEl(focusedSectionId, { scroll: true });
+      }
+    }
```

Comment:
When expanding there's no reason to change focus, right?

```suggestion
```

Reply:
@jonatanklosko but we need the scroll feature when the section expands out of the window. Since we no longer need expandAll collapseAll, I merged everything into a single function

Reply:
Ah that makes sense, shouldn't we scroll to `this.focusedId` instead though?

Reply:
Ah, actually it doesn't matter because we only expand all if everything is collapsed, so by definition a cell can't be focused.